sputnikdao2/src/policy.rs

```diff
@@ -237,8 +237,52 @@ impl VersionedPolicy {
 }
 
 impl Policy {
-    ///
-    /// Doesn't fail, because will be used on the finalization of the proposal.
+    pub fn add_or_update_role(&mut self, role: &RolePermission) {
+        for i in 0..self.roles.len() {
+            if &self.roles[i].name == &role.name {
+                env::log_str(&format!(
+                    "Updating existing role in the policy:{}",
+                    &role.name
+                ));
+                let _ = std::mem::replace(&mut self.roles[i], role.clone());
+                return;
+            }
+        }
+        env::log_str(&format!("Adding new role to the policy:{}", &role.name));
+        self.roles.push(role.clone());
+    }
+
+    pub fn remove_role(&mut self, role: &String) {
+        for i in 0..self.roles.len() {
```

Comment:
Could probably use `retain` here:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=1e84ddd396fd97ea7ab5d828e0de716a

Reply:
haha nice one. Could use that, but then I would need to do some extra checking to verify if the role really existed in the first place so I can print some error otherwise.